### PR TITLE
Angular renderer for object arrays

### PR DESCRIPTION
**Pull request by AlexandraBuzila**
Add array layout renderer for objects.

### CODE REVIEW

File: packages/angular-material/src/layouts/array-layout.renderer.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                Remove
+              </button>
+            </mat-card-actions>
+          </mat-card>
+        </div>
+      </div>
+    `,
+    changeDetection: ChangeDetectionStrategy.OnPush
+  })
+  export class ArrayLayoutRenderer extends JsonFormsBaseRenderer<ControlElement>
```
Comment (eneufeld) [inline packages/angular-material/src/layouts/array-layout.renderer.ts:92]: why not extend the JsonFormsArrayControl?

File: packages/angular-material/src/layouts/array-layout.renderer.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    template: `
+      <div fxLayout="column" fxLayoutGap="16px" [fxHide]="hidden">
+        <mat-toolbar>
+          <h2>{{ this.label }}</h2>
+          <span fxFlex></span>
+          <button
+            mat-icon-button
+            [disabled]="readonly"
+            (click)="add()"
+            attr.aria-label="{{ 'Add to ' + this.label + 'button' }}"
```
Comment (eneufeld) [inline packages/angular-material/src/layouts/array-layout.renderer.ts:60]: we will need to find a localization solution

File: packages/angular-material/src/layouts/array-layout.renderer.ts (hunk 1 above)
Comment (AlexandraBuzila) [reply packages/angular-material/src/layouts/array-layout.renderer.ts]: Good point! I changed the renderer so it extends `JsonFormsArrayControl`

File: packages/angular-material/src/layouts/array-layout.renderer.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  get defaultUISchema() {
+    if (!this._uischema) {
+      this._uischema = generateDefaultUISchema(this.scopedSchema);
+    }
+    return this._uischema;
+  }
+
+  getProps(index: number): OwnPropsOfRenderer {
+    const uischema =
+      (this.uischema.options?.detailUISchema as UISchemaElement) ??
```
Comment (eneufeld) [inline packages/angular-material/src/layouts/array-layout.renderer.ts:124]: we should unify this with the react renderer and use `detail`

Comment (eneufeld) [reply packages/angular-material/src/layouts/array-layout.renderer.ts:124]: or even better use the findschema like here: https://github.com/eclipsesource/jsonforms/blob/6cf39efbb82fdf77b173c38ca4b706d526920f9b/packages/material/src/layouts/ExpandPanelRenderer.tsx#L101

File: packages/angular-material/src/layouts/array-layout.renderer.ts [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    };
+  }
+  trackByFn(index: number) {
+    return index;
+  }
+}
+
+export const ArrayLayoutRendererTester: RankedTester = rankWith(
+  4,
+  isObjectArray
```
Comment (eneufeld) [inline packages/angular-material/src/layouts/array-layout.renderer.ts:147]: shouldn't we check for `isObjectArrayWithNesting` instead?

Comment (AlexandraBuzila) [reply packages/angular-material/src/layouts/array-layout.renderer.ts:147]: Isn't that tester too restrictive? If we don't have a nested array or don't provide a detail schema in the options, then the new renderer won't be used.

File: packages/angular-material/src/layouts/array-layout.renderer.ts [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+export class ArrayLayoutRenderer extends JsonFormsArrayControl {
+  noData: boolean;
+  addItem: (path: string, value: any) => () => void;
+  removeItems: (path: string, toDelete: number[]) => () => void;
+  uischemas: {
+    tester: UISchemaTester;
+    uischema: UISchemaElement;
+  }[];
+
+  constructor(private jsonFormAngularService: JsonFormsAngularService) {
```
Comment (eneufeld) [inline packages/angular-material/src/layouts/array-layout.renderer.ts:96]: we don't need to make this private as we can access it from the parent

Comment (eneufeld) [reply packages/angular-material/src/layouts/array-layout.renderer.ts:96]: If not, it should be accessible via the parent

**Comment by eneufeld** [comment]
Do you know why neither the buttons nor the toolbar is using the default styling colors of material?

File: packages/angular-material/src/layouts/array-layout.renderer.ts (hunk 5 above)
Comment (AlexandraBuzila) [reply packages/angular-material/src/layouts/array-layout.renderer.ts]: Fixed

**Comment by AlexandraBuzila** [comment]
I did the small refactoring the the array renderer.

Regarding the styling:
- the button style was not consistent with the other renderers, I fixed that
- the toolbar styling was a little different because we were not using the Roboto font.  I added it to the Angular Material examples, but I can revert this change if you don't agree with it.
The toolbar should look now like in the Angular Material documentation: https://material.angular.io/components/toolbar/overview

**Comment by AlexandraBuzila** [comment]
Ah, and the toolbar color is not set to 'primary'. I'll update my commit to use it.

**Pull request merged**